movie/views.py

```diff
@@ -27,4 +27,92 @@ def get(self, request):
             return JsonResponse({'result': movie_id}, status=200)
         except Movie.DoesNotExist:
             return JsonResponse({'message': 'Movie Does Not Exist'}, status=400)
+
+
+class MovieView(View):
+    def get(self, request, movie_id):
+        try:
+            movie = Movie.objects.get(id=movie_id)
+            stars = UserStar.objects.filter(movie=movie)
+            
+            panel_list = [
+                            {
+                                'role'     : panel.role,    
+                                'character': panel.character, 
+                                'actor'    : panel.actor.name, 
+                                'profile'  : panel.actor.image_url
+                            }
+                            for panel in Panel.objects.filter(movie_id=movie_id).select_related('actor')
+                        ]
+
+            star_graph = [{'x': 5, 'y': 0},
+                          {'x': 4, 'y': 0},
+                          {'x': 3, 'y': 0},
+                          {'x': 2, 'y': 0},
+                          {'x': 1, 'y': 0}]
+
+            for user_star in stars:
+                for star in star_graph:
+                    if star['x'] == user_star.star:
+                        star['y'] += 1
+                        
+            comment_list = [
+                {
+                    'user'     : comment.user.name,
+                    'user_star': UserStar.objects.filter(user=comment.user, movie_id=movie_id).first().star 
+                                if UserStar.objects.filter(user=comment.user, movie_id=movie_id).exists() else None,
+                    'content'  : comment.content,
+                    'like'     : comment.commentlike_set.all().count()
+                }
+                for comment in Comment.objects.filter(movie_id=movie_id).select_related('user')
+            ]
+            
+            same_genre_list = []
+            for moviegenre_movies in MovieGenre.objects.filter(movie_id=movie_id):
+                for genre in MovieGenre.objects.filter(genre=moviegenre_movies.genre):
+                    genre_stars = UserStar.objects.filter(movie=genre.movie)
```

Comment:
꼭 필요한 경우를 제외하고, 이중 반복문은 권장하지 않습니다! 어떠한 경우를 위한 코드인지 확인부탁드립니다 :0

Reply:
해당 영화의 장르들(복수)이 같은 영화들을 뿌리는 리스트입니다!
1) movie_genres 테이블에서 우선 해당 영화의 장르를 확인 후,
2) 그 장르가 같은 영화들을 찾는 코드입니다!
혹시 이중 포문을 쓰지 않는 좋은 방법이 있을까요..?
![IMG_8466](https://user-images.githubusercontent.com/69490984/113798094-7855bd80-978d-11eb-97f0-8ad2d2e09617.jpg)

Reply:
```suggestion
            MovieGenre.objects.filter(genre_id=movie.genre_id)
```

이미 movie를 담고 있는, 그 movie 에서 genre_id 를 필터하면 해당 영화와 같은 genre 의 movie genre 데이터를 가져 올 것 같네요! 위 코드는 예시이니 아이디어만 체크해서 코드 작성해보시고, star 도 확인해보세요!

Reply:
@kst6294  아 헐,,, 그러네요!!
그 부분을 생각못했네요!

감사합니다!!

Reply:
@kst6294 
장르가 여러 개일 경우가 있어서 for문 하나로는 힘들 것 같고, 장고 filter(genre_id__in=<리스트>)를 통해 for문을 분리했습니다!